project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -119,9 +132,10 @@ public String getAllAssignmentsJSON() {
 
   public String getAllQuestionsJSON(String key) {
     // query all questions with this key at assignment_id key
-    if (key != null) {
+    try {
         Filter propertyFilter = new FilterPredicate("assignmentKey", FilterOperator.EQUAL, key);
         Query query = new Query("Question").setFilter(propertyFilter);
+
         PreparedQuery results = this.queryDatabase(query);
         List<Question> questions = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
```

Comment:
This can be `for (Entity entity : this.query(query)) {` now

Reply:
will change